src/stores/pacbioPoolCreate.js

```diff
@@ -0,0 +1,954 @@
+      const key = `_${library.pacbio_request_id}`
+      this.libraries[key] = { ...this.libraries[key], ...library }
+    },
+
+    /**
+     * Deselects a plate and all its contents (wells and requests) based on the provided plate ID.
+     * Iterates over all wells of the plate and for each well, iterates over all requests and deselects them.
+     *
+     * @param {number} plateId - The ID of the plate to deselect.
+     * @returns {void}
+     *
+     * @example
+     * // Deselect a plate and its contents
+     * deselectPlateAndContents(1)
+     */
+    deselectPlateAndContents(plateId) {
+      this.selectPlate({ id: plateId, selected: false })
+      const { wells } = this.resources.plates[plateId]
+      for (const wellId of wells) {
+        const { requests = [] } = this.resources.wells[wellId]
+        for (const requestId of requests) {
+          this.selectRequest({ id: requestId, selected: false })
+        }
+      }
+    },
+
+    /**
+     * Deselects a tube and all its contents (requests) based on the provided tube barcode.
+     * If the tube cannot be found, the function returns without doing anything.
+     * If the tube can be found, deselects the tube and then iterates over all requests of the tube and deselects them.
+     *
+     * @param {string} tubeBarcode - The barcode of the tube to deselect.
+     * @returns {void}
+     *
+     * @example
+     * // Deselect a tube and its contents
+     * deselectTubeAndContents('TRAC-123')
+     */
+    deselectTubeAndContents(tubeBarcode) {
+      const tube = Object.values(this.resources.tubes).find((tube) => tube.barcode == tubeBarcode)
+      if (!tube) return
+      this.selectTube({ id: tube.id, selected: false })
+      const { requests } = this.resources.tubes[tube.id]
+      for (const requestId of requests) {
+        this.selectRequest({ id: requestId, selected: false })
+      }
+    },
+
+    /**
+     * Asynchronously creates a pool with the given libraries and pools.
+     * Validates the libraries before proceeding.
+     * If the libraries are not valid, returns an error response.
+     * Otherwise, sends a request to create the pool and handles the response.
+     * Groups the included resources by type and extracts the barcode from the first tube.
+     *
+     * @async
+     * @returns {Promise<Object>} A promise that resolves to an object containing the success status, barcode, and any errors.
+     *
+     * @example
+     * // Create a pool
+     * const result = await createPool();
+     * console.log(result); // { success: true, barcode: 'barcode123', errors: [] }
+     */
+    async createPool() {
+      const { libraries, pool } = this
+      if (!validate(libraries)) return { success: false, errors: 'The pool is invalid' }
+      const rootStore = useRootStore()
+      const request = rootStore.api.traction.pacbio.pools
+      const promise = request.create({
```

Comment:
Is there a plan here to feature flag between sending used_aliquots vs libraries?